common/openapi.yml

```diff
@@ -287,6 +287,17 @@ components:
             type: object
       description: List of catalog datasets
 
+    DatasetAccessLevel:
+      type: string
+      description: The level of access a user or group has on a Dataset. Owners can create and delete
+        dataset entries, writers can modify only, readers can read the data, and discoverers can see
```

Comment:
I don't think the catalog needs to distinguish between `writer` and `owner`. Can you describe a case where we'd want to allow a user to modify the contents of a catalog entry but not create or delete it? If a user can overwrite all elements in an entry, they could replace it with an empty one, which would have the same effect as deletion.

My thinking is that we'd want to map workspace writers to the `reader` role, as we'd want to limit catalog entry creation to the owner of the data, not a user who has write access to the workspace.